src/octoprint/filemanager/analysis.py

```diff
@@ -384,7 +387,7 @@ def _do_analysis(self, high_priority=False):
 			if analysis["extrusion_length"]:
 				result["filament"] = dict()
 				for i in range(len(analysis["extrusion_length"])):
-					result["filament"]["tool%d" % i] = {
+					result["filament"]["tools"][i] = {
```

Comment:
Please do not change the internal format used here. It might already be used by some plugins and by changing this you are making things backwards incompatible. It's also used in the frontend as-is, so this change already breaks filament usage display there.

Reply:
I understand the retrocompatibility issue, but you can't iterrate safely on its. maybe we could add (instead to replace) the new entry "tools" and set old "tool%i" as deprecated.

This is used by history plugin for example (it use only "tool1" and "tool2"...), but I made a PR too.

Reply:
Huh? You can iterate perfectly fine over the keys in this dict. I don't understand why you want to change this, there is no reason (especially not within the context of this PR) and it only produces compatibility issues.